colossalai/nn/layer/parallel_2d/_operation.py

```diff
@@ -730,6 +730,12 @@ def split_tensor_2d(input_: Tensor, dim: int = 0) -> Tensor:
     Returns:
         :class:`torch.tensor`: The tensor has been split.
     """
+    dim_size = input_.size(dim)
+    world_size = gpc.get_world_size(ParallelMode.PARALLEL_2D_COL)
+    assert dim_size % world_size == 0, \
+        f'The dimension {dim} to split, size ({dim_size}) is not a multiple of world size ({world_size}), ' \
```

Comment:
This is splitting batch size. So just say the batch size is not a multiple of 2D size / 2.5D size * depth / square of 3D depth